packages/firestore/src/local/local_store_bundle.ts

```diff
@@ -0,0 +1,261 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { newQueryForPath, queryToTarget } from '../core/query';
+import { SnapshotVersion } from '../core/snapshot_version';
+import { Target } from '../core/target';
+import {
+  DocumentKeySet,
+  documentKeySet,
+  documentVersionMap,
+  maybeDocumentMap,
+  MaybeDocumentMap
+} from '../model/collections';
+import { debugCast } from '../util/assert';
+
+import {
+  BundleMetadata,
+  NamedQuery as ProtoNamedQuery
+} from '../protos/firestore_bundle_proto';
+import { BundleConverter, BundledDocuments } from '../core/bundle';
+import { fromVersion } from '../remote/serializer';
+import { fromBundledQuery } from './local_serializer';
+import { ByteString } from '../util/byte_string';
+import { ResourcePath } from '../model/path';
+import {
+  LocalStore,
+  LocalStoreImpl,
+  allocateTarget,
+  populateDocumentChangeBuffer
+} from './local_store';
+import { NamedQuery } from '../core/bundle_types';
+
+/**
+ * Applies the documents from a bundle to the "ground-state" (remote)
+ * documents.
+ *
+ * LocalDocuments are re-calculated if there are remaining mutations in the
+ * queue.
+ */
+export async function applyBundleDocuments(
```

Comment:
Would prefer to have this back in local_store.ts

Reply:
Done.